### PR TITLE
Add remove for part log

### PR DESCRIPTION
**Pull request by Ravengg**

### CODE REVIEW

**Comment by robot-metrika-test** [comment]
Can one of the admins verify this patch?

File: dbms/src/Storages/MergeTree/MergeTreeDataWriter.cpp [hunk 1]
```diff
@@ -168,13 +166,16 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataWriter::writeTempPart(BlockWithDa
...
+		elem.event_time = time(0);
+
 		elem.event_type = PartLogElement::NEW_PART;
 		elem.size_in_bytes = new_data_part->size_in_bytes;
 		elem.duration_ms = stopwatch.elapsed() / 1000000;
 
 		elem.database_name = new_data_part->storage.getDatabaseName();
 		elem.table_name = new_data_part->storage.getTableName();
-		elem.part_name = new_data_part->name;
+		elem.part_name = new_data_part->name.substr(4);
```
Comment (alexey-milovidov) [inline dbms/src/Storages/MergeTree/MergeTreeDataWriter.cpp]: Very unobvious.
`strlen("tmp_")` should be better, but also not good enough.
Keeping separate variables for temporary name and result name will be better.

File: dbms/src/Storages/StorageMergeTree.cpp [hunk 2]
```diff
@@ -364,6 +364,17 @@ bool StorageMergeTree::merge(
 		elem.duration_ms = stopwatch.elapsed() / 1000000;
 
 		part_log->add(elem);
+
+		elem.duration_ms = 0;
+		elem.event_type = PartLogElement::REMOVE_PART;
+		elem.merged_from = Strings();
```
Comment (alexey-milovidov) [inline dbms/src/Storages/StorageMergeTree.cpp]: This is redundant.

File: dbms/src/Storages/StorageReplicatedMergeTree.cpp [hunk 3]
```diff
@@ -2103,6 +2098,40 @@ bool StorageReplicatedMergeTree::fetchPart(const String & part_name, const Strin
 		MergeTreeData::Transaction transaction;
 		auto removed_parts = data.renameTempPartAndReplace(part, nullptr, &transaction);
 
+		std::shared_ptr<PartLog> part_log = context.getPartLog();
+		if (part_log)
```
Comment (alexey-milovidov) [inline dbms/src/Storages/StorageReplicatedMergeTree.cpp]: BTW, you could write 
`if (std::shared_ptr<PartLog> part_log = context.getPartLog())`
and avoid
`part_log.reset();`

**Pull request merged**